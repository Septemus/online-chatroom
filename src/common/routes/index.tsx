import App from "@/content/App";
import Account from "@/content/components/account";
import Chatroom from "@/content/components/chatroom";
import LoginPage from "@/content/components/loginPage";
import LoginForm from "@/content/components/loginPage/loginForm";
import RegisterForm from "@/content/components/loginPage/registerForm";
import { createRoutesFromElements, Navigate, Route } from "react-router-dom";
import registerAction from "../apollo/client/queries/register";

import AccountPosts from "@/content/components/account/posts";
import AccountPhotos from "@/content/components/account/photos";
import Options from "@/content/components/options";
import EditProfile from "@/content/components/options/editProfile";
import loginAction from "../apollo/client/queries/login";

export default createRoutesFromElements(
	<>
		<Route
			element={<App />}
			path="/"
		>
			<Route
				element={<Chatroom />}
				path="chatroom"
			></Route>
			<Route
				element={<Account />}
				path="account"
			>
				<Route
					index
					element={
						<Navigate
							to="posts"
							replace
						></Navigate>
					}
				></Route>
				<Route
					path="posts"
					element={<AccountPosts />}
				></Route>
				<Route
					path="photos"
					element={<AccountPhotos />}
				></Route>
			</Route>
			<Route
				element={<Options />}
				path="options"
			>
				<Route
					index
					element={
						<Navigate
							to="editProfile"
							replace
						></Navigate>
					}
				></Route>
				<Route
					element={<EditProfile />}
					path="editProfile"
				></Route>
			</Route>
		</Route>
		<Route
			element={<LoginPage />}
			path="/login"
			action={loginAction}
		>
			<Route
				element={<LoginForm />}
				path=""
			></Route>
			<Route
				element={<RegisterForm />}
				path="register"
				action={registerAction}
			></Route>
		</Route>
	</>,
);
