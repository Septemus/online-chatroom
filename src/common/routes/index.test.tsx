import routes from ".";
test("routes stable", () => {
	expect(routes).toMatchSnapshot();
});
