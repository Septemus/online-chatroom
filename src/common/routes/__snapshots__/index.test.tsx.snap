// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes stable 1`] = `
Array [
  Object {
    "Component": undefined,
    "ErrorBoundary": undefined,
    "action": undefined,
    "caseSensitive": undefined,
    "children": Array [
      Object {
        "Component": undefined,
        "ErrorBoundary": undefined,
        "action": undefined,
        "caseSensitive": undefined,
        "children": Array [
          Object {
            "Component": undefined,
            "ErrorBoundary": undefined,
            "action": undefined,
            "caseSensitive": undefined,
            "element": <Chatbox />,
            "errorElement": undefined,
            "handle": undefined,
            "hasErrorBoundary": false,
            "id": "0-0-0-0",
            "index": undefined,
            "lazy": undefined,
            "loader": undefined,
            "path": "chatbox",
            "shouldRevalidate": undefined,
          },
        ],
        "element": <Message />,
        "errorElement": undefined,
        "handle": undefined,
        "hasErrorBoundary": false,
        "id": "0-0-0",
        "index": undefined,
        "lazy": undefined,
        "loader": undefined,
        "path": "message",
        "shouldRevalidate": undefined,
      },
      Object {
        "Component": undefined,
        "ErrorBoundary": undefined,
        "action": undefined,
        "caseSensitive": undefined,
        "children": Array [
          Object {
            "Component": undefined,
            "ErrorBoundary": undefined,
            "action": undefined,
            "caseSensitive": undefined,
            "element": <Navigate
              replace={true}
              to="posts"
            />,
            "errorElement": undefined,
            "handle": undefined,
            "hasErrorBoundary": false,
            "id": "0-0-1-0",
            "index": true,
            "lazy": undefined,
            "loader": undefined,
            "path": undefined,
            "shouldRevalidate": undefined,
          },
          Object {
            "Component": undefined,
            "ErrorBoundary": undefined,
            "action": undefined,
            "caseSensitive": undefined,
            "element": <AccountPosts />,
            "errorElement": undefined,
            "handle": undefined,
            "hasErrorBoundary": false,
            "id": "0-0-1-1",
            "index": undefined,
            "lazy": undefined,
            "loader": undefined,
            "path": "posts",
            "shouldRevalidate": undefined,
          },
          Object {
            "Component": undefined,
            "ErrorBoundary": undefined,
            "action": undefined,
            "caseSensitive": undefined,
            "element": <AccountPhotos />,
            "errorElement": undefined,
            "handle": undefined,
            "hasErrorBoundary": false,
            "id": "0-0-1-2",
            "index": undefined,
            "lazy": undefined,
            "loader": undefined,
            "path": "photos",
            "shouldRevalidate": undefined,
          },
        ],
        "element": <Account />,
        "errorElement": undefined,
        "handle": undefined,
        "hasErrorBoundary": false,
        "id": "0-0-1",
        "index": undefined,
        "lazy": undefined,
        "loader": undefined,
        "path": "account",
        "shouldRevalidate": undefined,
      },
      Object {
        "Component": undefined,
        "ErrorBoundary": undefined,
        "action": undefined,
        "caseSensitive": undefined,
        "children": Array [
          Object {
            "Component": undefined,
            "ErrorBoundary": undefined,
            "action": undefined,
            "caseSensitive": undefined,
            "element": <Navigate
              replace={true}
              to="editProfile"
            />,
            "errorElement": undefined,
            "handle": undefined,
            "hasErrorBoundary": false,
            "id": "0-0-2-0",
            "index": true,
            "lazy": undefined,
            "loader": undefined,
            "path": undefined,
            "shouldRevalidate": undefined,
          },
          Object {
            "Component": undefined,
            "ErrorBoundary": undefined,
            "action": [Function],
            "caseSensitive": undefined,
            "element": <EditProfile />,
            "errorElement": undefined,
            "handle": undefined,
            "hasErrorBoundary": false,
            "id": "0-0-2-1",
            "index": undefined,
            "lazy": undefined,
            "loader": undefined,
            "path": "editProfile",
            "shouldRevalidate": undefined,
          },
        ],
        "element": <Options />,
        "errorElement": undefined,
        "handle": undefined,
        "hasErrorBoundary": false,
        "id": "0-0-2",
        "index": undefined,
        "lazy": undefined,
        "loader": undefined,
        "path": "options",
        "shouldRevalidate": undefined,
      },
    ],
    "element": <App />,
    "errorElement": undefined,
    "handle": undefined,
    "hasErrorBoundary": false,
    "id": "0-0",
    "index": undefined,
    "lazy": undefined,
    "loader": undefined,
    "path": "/",
    "shouldRevalidate": undefined,
  },
  Object {
    "Component": undefined,
    "ErrorBoundary": undefined,
    "action": [Function],
    "caseSensitive": undefined,
    "children": Array [
      Object {
        "Component": undefined,
        "ErrorBoundary": undefined,
        "action": undefined,
        "caseSensitive": undefined,
        "element": <LoginForm />,
        "errorElement": undefined,
        "handle": undefined,
        "hasErrorBoundary": false,
        "id": "0-1-0",
        "index": undefined,
        "lazy": undefined,
        "loader": undefined,
        "path": "",
        "shouldRevalidate": undefined,
      },
      Object {
        "Component": undefined,
        "ErrorBoundary": undefined,
        "action": [Function],
        "caseSensitive": undefined,
        "element": <RegisterForm />,
        "errorElement": undefined,
        "handle": undefined,
        "hasErrorBoundary": false,
        "id": "0-1-1",
        "index": undefined,
        "lazy": undefined,
        "loader": undefined,
        "path": "register",
        "shouldRevalidate": undefined,
      },
    ],
    "element": <LoginPage />,
    "errorElement": undefined,
    "handle": undefined,
    "hasErrorBoundary": false,
    "id": "0-1",
    "index": undefined,
    "lazy": undefined,
    "loader": undefined,
    "path": "/login",
    "shouldRevalidate": undefined,
  },
]
`;
